tests/TaskLineRenderer.test.ts

```diff
@@ -0,0 +1,156 @@
+    return textSpan;
+}
+
+function getDescriptionText(parentElement: HTMLElement) {
+    const textSpan = getTextSpan(parentElement);
+    return textSpan.innerText;
+}
+
+describe('task line rendering', () => {
+    it('creates the correct span structure for a basic task', async () => {
+        const taskLine = '- [ ] This is a simple task';
+        const task = fromLine({
+            line: taskLine,
+        });
+        const parentRender = await createMockParentAndRender(task);
+
+        // Check what we have one child, which is the rendered child
+        expect(parentRender.children.length).toEqual(1);
+        const li = parentRender.children[0];
+        // Check that it's an element of type LI
+        expect(li.nodeName).toEqual('LI');
+        // Check that it has two children: a checkbox and a text span
+        expect(li.children.length).toEqual(2);
+        const checkbox = li.children[0];
+        expect(checkbox.nodeName).toEqual('INPUT');
+        expect(checkbox.classList.contains('task-list-item-checkbox')).toBeTruthy();
+        const textSpan = li.children[1];
+        expect(textSpan.nodeName).toEqual('SPAN');
+        expect(textSpan.classList.contains('tasks-list-text')).toBeTruthy();
+        expect((textSpan as HTMLSpanElement).innerText).toEqual('This is a simple task');
+    });
+
+    it('hides the global filter if and only if required', async () => {
+        const getDescriptionTest = async () => {
+            const taskLine = '- [ ] This is a simple task with a #global filter';
+            const task = fromLine({
+                line: taskLine,
+            });
+            const parentRender = await createMockParentAndRender(task);
+            return getDescriptionText(parentRender);
+        };
+
+        const descriptionWithFilter = await getDescriptionTest();
+        expect(descriptionWithFilter).toEqual('This is a simple task with a #global filter');
+
+        updateSettings({ globalFilter: '#global', removeGlobalFilter: true });
+        const descriptionWithoutFilter = await getDescriptionTest();
+        expect(descriptionWithoutFilter).toEqual('This is a simple task with a  filter');
+        resetSettings();
+    });
+
+    it('renders task components according to the given layout', async () => {
+        const testLayoutOptions = async (
+            taskLine: string,
+            layoutOptions: Partial<LayoutOptions>,
+            expectedRender: string,
+        ) => {
+            const task = fromLine({
+                line: taskLine,
+            });
+            const fullLayoutOptions = { ...new LayoutOptions(), ...layoutOptions };
+            const parentRender = await createMockParentAndRender(task, fullLayoutOptions);
+            const renderedDescription = getDescriptionText(parentRender);
+            expect(renderedDescription).toEqual(expectedRender);
+        };
+
+        const taskLine = '- [ ] Wobble ⏫ 📅 2022-07-02 ⏳ 2022-07-02 🛫 2022-07-02 🔁 every day';
```

Comment:
Tests can be made more robust and useful by removing duplication in the input data.
For example, if someone broke the rendering of due date and made it write scheduled date instead, these tests would still pass.

I suggest using unique values for each of the dates in the initial task.

Reply:
Will do.